infrastructure/platform-apis/azuresqldatabase.yaml

```diff
@@ -0,0 +1,128 @@
+apiVersion: kro.run/v1alpha1
+kind: ResourceGraphDefinition
+metadata:
+  name: azure-sql-database
+  labels:
+    homelab.yeseh.nl/llm-exposed: "true"
+    homelab.yeseh.nl/api-category: "databases"
+  annotations:
+    homelab.yeseh.nl/llm-description: |
+      The AzureSqlDatabase resource allows you to create an Azure SQL Database with a private endpoint.
+      It links to an existing Canvas for permissions and networking.
+spec:
+  schema:
+    apiVersion: v1alpha1
+    kind: AzureSqlDatabase
+    spec:
+      # Canvas linking
+      canvasName: string | required=true
+      canvasEnvironment: string | required=true
+      
+      # SQL Server settings
+      serverName: string | required=true
+      location: string | required=true
+      administratorLogin: string | required=true
+      administratorLoginPassword: string | required=true
+      
+      # Database settings
+      databaseName: string | required=true
+      skuName: string | default="Basic"
+      tier: string | default="Basic"
+      maxSizeBytes: integer | default=1073741824 # 1GB
+      
+      # Private endpoint settings
+      enablePrivateEndpoint: boolean | default=true
+      privateEndpointSubnetId: string | required=true
+      
+    status:
+      serverResourceId: ${sqlServer.status.id}
+      databaseResourceId: ${sqlDatabase.status.id}
+      privateEndpointResourceId: ${privateEndpoint.status.id}
+      fqdn: ${sqlServer.status.fullyQualifiedDomainName}
+      connectionString: ${sqlServer.status.fullyQualifiedDomainName}
+
+  resources:
+    # Get canvas information for reference
+    - id: canvas
```

Comment:
Pass the canvas managed identity in as an existing resource. Do not create it here

Reply:
Existing is not a valid property, undo this change